enumerables.rb

```diff
@@ -0,0 +1,150 @@
+# rubocop:disable Metrics/ModuleLength
+module Enumerable
+  def my_each(&block)
+    return to_enum(:my_each) unless block_given?
+
+    to_a.each(&block)
+    self
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    (0...to_a.length).each do |i|
+      yield(to_a[i], i)
+    end
+    self
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    my_array = []
+    my_each do |item|
+      my_array.push(item) if yield(item)
+    end
+    my_array
+  end
+
+  # rubocop:disable Style/CaseEquality
+  def my_all?(*arg)
+    result = true
+    if !arg[0].nil?
+      my_each do |item|
+        result = false unless arg[0] === item
+      end
+
+    elsif !block_given?
+      result = true
+      my_each do |item|
+        result = false unless item
+      end
+    else
+      result = true
+      my_each do |item|
+        result = false unless yield(item)
+      end
+    end
+    result
+  end
+
+  def my_any?(*arg)
+    result = false
+    if !arg[0].nil?
+      my_each do |item|
+        result = true if arg[0] === item
+      end
+    elsif !block_given?
+      result = false
+      my_each do |item|
+        result = true if item
+      end
+    else
+      result = false
+      my_each do |item|
+        result = true if yield(item)
+      end
+    end
+    result
+  end
+
+  def my_none?(*arg)
+    result = true
+    if !arg[0].nil?
+      my_each do |item|
+        result = false if arg[0] === item
+      end
+
+    elsif !block_given?
+      result = true
+      my_each do |item|
+        result = false if item
+      end
+
+    else
+      result = true
+      my_each do |item|
+        result = false if yield(item)
+      end
+    end
+    result
+  end
+
+  def my_count(arg = nil)
+    match = 0
+    if arg
+      my_each do |item|
+        match += 1 if item == arg
+      end
+
+    elsif block_given?
+      my_each do |item|
+        match += 1 if yield item
+      end
+
+    else
+      match = to_a.length
+    end
+
+    match
+  end
+
+  # rubocop:enable Style/CaseEquality
+  def my_map(proc = nil)
+    # rubocop:disable Lint/ToEnumArguments
+    return to_enum(:my_map) unless block_given?
+    # rubocop:enable Lint/ToEnumArguments
+
+    new_arr = []
+    if proc
+      my_each do |item|
+        new_arr.push(proc.call(item))
+      end
+    else
+      my_each do |item|
+        new_arr.push(yield item)
+      end
+    end
+    new_arr
+  end
+
+  # rubocop:disable Metrics/PerceivedComplexity,Metrics/CyclomaticComplexity
+  def my_inject(acc_value = nil, value_two = nil)
+    if (!acc_value.nil? && value_twog.nil?) && (acc_value.is_a?(symbol) || acc_value.is_a?(string))
+      value_two = acc_value
```

Comment:
##  `my_inject`

- [ ] when a block is given with an initial value combines all elements of enum by applying a binary operation, specified by a block
```
Failure/Error: expect(array.my_inject(20, :*)).to eq array.inject(20, :*)
     
     NameError:
       undefined local variable or method `value_twog'
```
- You have a typo in there. It should be `value_two` and not `value_twog`.
- Give this a quick fix

- [ ] when a symbol is specified without an initial value combines each element of the collection by applying the symbol as a named method:range
```
Failure/Error: expect(range.my_inject(:*)).to eq range.inject(:*)
     
     NameError:
       undefined local variable or method `symbol' for 5..50:Range



Failure/Error: expect(array.my_inject(:+)).to eq array.inject(:+)
     
     NameError:
       undefined local variable or method `symbol'
```
- Your method doesn't seem to recognize these operation symbols. Give this a quick fix as well.

- [ ] when a block is given with an initial value combines all elements of enum by applying a binary operation, specified by a block
```
Failure/Error: actual = range.my_inject(4) { |prod, n| prod * n }
     
     NameError:
       undefined local variable or method `symbol' for 5..50:Range

```
- Give this section a fix as well.

Reply:
Thanks for the review @mke2111, I will implement the fix